### PR TITLE
Consolidate subscribers

### PR DESCRIPTION
**Pull request by jon-betts**
Before looking at changing how we react to annotation events I wanted to workout how we are currently using them. This meant trying to understand them, at which point I found out they are pretty heavily strewn all over the place. 

This PR pulls all of the subscribers and the definitions of what events they listen to into the same `subscribers.py` file.

This does not touch the subscriber in the websocket.

### CODE REVIEW

File: h/nipsa/subscribers.py [hunk 1]
```diff
@@ -1,18 +1,12 @@
...
-
-    nipsa = _user_nipsa(event.request, payload)
-
-    if nipsa:
-        payload["nipsa"] = True
-
-
-def _user_nipsa(request, payload):
-    nipsa_service = request.find_service(name="nipsa")
-    return "user" in payload and nipsa_service.is_flagged(payload["user"])
```
Comment (jon-betts) [inline h/nipsa/subscribers.py:13]: This stuff was in-lined, allowing an "optimisation" where we don't get the service if we aren't going to call it. Mostly to make the logic clearer.

File: h/nipsa/subscribers.py [hunk 2]
```diff
@@ -1,18 +1,12 @@
...
-
-
-def _annotation_moderated(request, annotation):
-    svc = request.find_service(name="annotation_moderation")
-    return svc.hidden(annotation)
```
Comment (jon-betts) [inline h/nipsa/subscribers.py:18]: This didn't appear to do anything, so it's just gone.

File: tests/h/nipsa/subscribers_test.py [hunk 3]
```diff
@@ -1,77 +1,43 @@
...
+from unittest.mock import create_autospec
 
 import pytest
 
-from h.nipsa import subscribers
+from h.events import AnnotationTransformEvent
+from h.nipsa.subscribers import nipsa_transform_annotation
+from h.services.nipsa import NipsaService
 
-FakeEvent = namedtuple("FakeEvent", ["request", "annotation", "annotation_dict"])
```
Comment (jon-betts) [inline tests/h/nipsa/subscribers_test.py:8]: The real event is a named tuple. I can't see what's gained by replicating it.

File: tests/h/nipsa/subscribers_test.py [hunk 4]
```diff
@@ -1,77 +1,43 @@
...
 
-
-class FakeAnnotation:
-    def __init__(self, data):
-        self.data = data
-
-    @property
-    def id(self):
-        return self.data["id"]
```
Comment (jon-betts) [inline tests/h/nipsa/subscribers_test.py:17]: Replaced with factory usage. We don't actually use any fields (including id)

File: h/app.py [hunk 5]
```diff
@@ -39,16 +39,7 @@ def includeme(config):
...
-
-    config.add_subscriber(
-        "h.subscribers.add_renderer_globals", "pyramid.events.BeforeRender"
-    )
-    config.add_subscriber(
-        "h.subscribers.publish_annotation_event", "h.events.AnnotationEvent"
-    )
-    config.add_subscriber(
-        "h.subscribers.send_reply_notifications", "h.events.AnnotationEvent"
-    )
```
Comment (seanh) [inline h/app.py:51]: :+1:

**Review by seanh — APPROVED** [review]
Looks good. I'm assuming you've tested that all these subscribers are still working!

**Comment by jon-betts** [comment]
You assume incorrectly. This is still pretty bust. I didn't want to go to town until I'd got the thumbs up it wasn't in general mental.

**Comment by jon-betts** [comment]
I can confirm these all get kicked off. In fact `add_renderer_globals` gets kicked off, _a lot_. I wonder if that's needed for all calls?

Anyway, I think this is working, I've fixed a small boo boo with the config.

**Pull request merged**